### ISSUE TITLE
Prompt should not appear in Interactive Window while script is running

### ISSUE THREAD

**Issue by theficus** [adamdriscoll/poshtools#224]
When running a script, the input prompt in the interactive view remains visible and allows for text entry that gets queued until the script is done running. This is not at all consistent with ISE that blocks all input until the script is running.

I've also found that if a background operation is running (for instance modules are being resolved or something like that), the prompt lets you enter commands which are then completely discarded.

If operations are happening, the interactive mode should block input like ISE does.

**Comment by AndreSayreMSFT** [comment]
Hi @theficus ! 

When running via non-Repl, eg, Right-clicking and selecting Execute from the Solution Explorer, is that what you're referring to?

**Comment by EricMSFT** [comment]
I think @theficus  is referring to the REPL window prompt, I agree that we should apply a readonly protection on prompt while we are in the middle of executing something.

**Comment by bmoore-msft** [comment]
yeah, we need to do some more thinking on the behavior of that thing during debugging...
